pycbc/events/coinc.py

```diff
@@ -612,11 +621,60 @@ def __init__(self, num_templates, analysis_block, background_statistic,
 
         self.singles = {}
 
-        #if save_background_on_interrupt:
-        #    import signal
-        #    def sig_handler(signum, frame):
-        #        pass
-        #    signal.signal(signal.SIGINT, sig_handler)
+    # Adding this here as it requires the results format dict as defined by this
+    # class. 
+    @classmethod
+    def pick_best_coinc(cls, coinc_results):
+        """Choose the best coinc by ifar first, then statistic if needed.
+
+        This function picks which of the available coincs to use. It chooses
+        first by ifar, and if all else is equal by statistic value. A trails
+        factor is applied if multiple types of coincs were possible at this
+        time.
+
+        Parameters
+        ----------
+        coinc_results: list of coinc result dicts
+            Dictionary by detector pair of coinc result dicts.
+
+        Returns
+        -------
+        best: coinc results dict
+            Dict containing the results as would be returned by this class.
+            If there is a coinc, this will contain the 'best' one.
+        """
+        # Choose best by ifar, if equal choose loudest statistic value
+        mstat = 0
+        mifar = 0
+        mresult = None
+
+        # record the trials factor from the possible coincs we could
+        # maximize over
+        trials = 0
+        for result in coinc_results:
+            # Check that a coinc was possible
+            if 'coinc_possible' in result:
+                trials += 1
+
+                # Check that a coinc exists
+                if 'foreground/ifar' in result:
+                    ifar = result['foreground/ifar']
```

Comment:
is this always exactly one IFAR value?

Reply:
Yes, by construction. There is only one coincidence per analysis stride (~ 8 seconds).